python/shotgun_desktop/upgrade_startup.py

```diff
@@ -45,26 +45,28 @@ def upgrade_startup(splash, sgtk, app_bootstrap):
     if not _supports_get_from_location_and_paths(sgtk):
         return False
 
+    sg = sgtk.get_authenticated_user().create_sg_connection()
+
     current_desc = sgtk.descriptor.create_descriptor(
-        sgtk.get_authenticated_user().create_sg_connection(),
+        sg,
         sgtk.descriptor.Descriptor.FRAMEWORK,
         get_location(app_bootstrap)
     )
 
-    # A Dev descriptor means there is nothing to update. Early out so that we don't show
-    # "Getting Shotgun Desktop updates...""
-    if current_desc.is_dev():
-        logger.info("Desktop startup using a dev descriptor, skipping update...")
-        return False
-
-    logger.debug("Testing for remote access.", current_desc)
+    logger.debug("Testing for remote access: %s", current_desc)
 
-    if not current_desc.has_remote_accesss():
+    if not current_desc.has_remote_access():
         logger.info("Could not update %r: remote access not available.", current_desc)
```

Comment:
Is this message meant to be for end users? In that case, maybe it could be tweaked to say something like "Couldn't contact the update server, your current configuration %s will be used"